nixos/modules/programs/lomiri.nix

```diff
@@ -0,0 +1,160 @@
+    ];
+
+    services.dbus.packages = with pkgs.lomiri; [
+      hfd-service
+      libusermetrics
+      lomiri-download-manager
+    ];
+
+    fonts.packages = with pkgs; [
+      # Applications tend to default to Ubuntu font
+      ubuntu_font_family
+    ];
+
+    # Copy-pasted
+    # TODO are all of these needed? just nice-have's? convenience?
+    hardware.opengl.enable = lib.mkDefault true;
+    fonts.enableDefaultPackages = lib.mkDefault true;
+    programs.dconf.enable = lib.mkDefault true;
+    programs.xwayland.enable = lib.mkDefault true;
+
+    services.accounts-daemon.enable = true;
+
+    services.ayatana-indicators = {
+      enable = true;
+      packages = (with pkgs; [
+        ayatana-indicator-datetime
+        ayatana-indicator-messages
+        ayatana-indicator-session
+      ]) ++ (with pkgs.lomiri; [
+        telephony-service
+      ]);
+    };
+
+    services.udisks2.enable = true;
+    services.upower.enable = true;
+    services.geoclue2.enable = true;
+    services.printing.enable = true;
+
+    services.gnome.evolution-data-server = {
+      enable = true;
+      plugins = with pkgs; [
+        #address-book-service
+      ];
+    };
+
+    services.telepathy.enable = true;
+
+    services.xserver.displayManager.lightdm.enable = lib.mkDefault true;
+    services.xserver.displayManager.lightdm.greeters.lomiri.enable = lib.mkDefault true;
+    services.xserver.displayManager.defaultSession = lib.mkDefault "lomiri";
+    services.xserver.displayManager.sessionPackages = with pkgs.lomiri; [ lomiri-session ];
+    services.xserver.updateDbusEnvironment = true;
+
+    environment.pathsToLink = [
+      # Required for installed Ayatana indicators to show up in Lomiri
+      "/share/ayatana"
+      # Registers which applications may send data to which other applications
+      "/share/content-hub/peers"
+      # Registers which applications handle dispatched URLs
+      "/share/lomiri-url-dispatcher/urls"
+      # Splash screens & other images for desktop apps launched via lomiri-app-launch
+      "/share/lomiri-app-launch"
+      # Try to get maliit stuff working
+      "/share/maliit/plugins"
+      # Data
+      "/share/sounds"
+      "/share/wallpapers"
+      "/share/locale" # lomiri-ui-toolkit picks up i18n from only 1 directory
+    ];
+
+    # Unconditionally run service that collects system-installed URL handlers before l-u-d
+    # TODO also run user-installed one?
+    systemd.user.services."lomiri-url-dispatcher-update-system-dir" = {
+      description = "Lomiri URL dispatcher system directory updater";
+      wantedBy = [ "lomiri-url-dispatcher.service" ];
+      before = [ "lomiri-url-dispatcher.service" ];
+      serviceConfig = {
+        Type = "oneshot";
+        ExecStart = "${pkgs.lomiri.lomiri-url-dispatcher}/libexec/lomiri-url-dispatcher/lomiri-update-directory /run/current-system/sw/share/lomiri-url-dispatcher/urls/";
+      };
+    };
+
+    systemd.tmpfiles.rules = [
```

Comment:
Please use the new tmpfiles syntax

Reply:
Please add references when you say stuff like this. If I was already aware of what the "new syntax" was then I wouldn't be submitting it like this.

The syntax of `systemd.tmpfiles.rules` itself didn't change, so I'm *assuming* you're referring to (linking these here for when I have time to rewrite it):

- new option for specifying tmpfile rules: https://github.com/NixOS/nixpkgs/pull/262731
- example migration to new API: https://github.com/NixOS/nixpkgs/pull/280373

Reply:
Avoided the tmpfiles usage here via other means, but I adjusted the one in the tests to use `tmpfiles.settings` instead.